duckduckgo/Android
# Implement role manager default browser dialog in welcome page

Task/Issue URL: https://app.asana.com/0/72649045549333/1199338275364983/f
Tech Design URL: 
CC: 

**Description**:
Use the role manager dialog during onboarding welcome page to prompt the user to select us as default browser

* Added the `DefaultRoleBrowserDialogExperiment` that create the intent for the role manager and use it in welcome page

* Skip the "default browser page" regardless of user selection


**Steps to test this PR**:
Note: Use variant `zu`

==Test case 1==
1. Device API >=29
2. Fresh install the app and launch it
3. In Welcome page press "let's do it"
4. Select ddg as default browser in the dialog
5. Verify app proceeds to main browser page
6. Verify pixel `m_db_s` with `fo` param set as true
7. Verify browser is set as default

==Test case 2==
1. Device API >=29
2. Fresh install the app and launch it
3. In Welcome page press "let's do it"
4. Press "Cancel" in the dialog
5. Verify app proceeds to main browser page
6. Verify pixel `m_db_ns` with `fo` param set as true
7. Verify browser is NOT set as default

==Test case 3==
1. Device API <29
2. Fresh install the app and launch it
3. In Welcome page press "let's do it"
4. Verify onboarding proceeds unmodified

==Test case 4==
1. Device with any API
2. Upgrade the app and launch it (onboarding should have been done already)
3. Verify onboarding is skipped and user taken to main browser page

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
